mgr/cephadm: mon public network updating

timqn22 opened the pull request:
Currently when a new public network is set for mon daemons and the mon daemons are restarted, the new public network is still not applied. This is because the public network config doesn't update dynamically and it requires the mon daemon to be redeployed for the change to take effect. People assume that restarting the mon daemon should apply the change so my code adds a function to check if the mon daemon public network config has been changed and has not been applied yet. This function is then called in _daemon_action to reroute a restart request to the redeploy request process, essentially turning the restart into a redeploy to update the public network.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2400397

Upstream Tracker: https://tracker.ceph.com/issues/73775

Signed-off-by: Timothy Q Nguyen <timqn22@gmail.com>

timqn22 committed:
mgr/cephadm: mon public network updating

Currently when a new public network is set for mon daemons and
the mon daemons are restarted, the new public network is still
not applied. This is because the public network config doesn't
update dynamically and it requires the mon daemon to be
redeployed for the change to take effect. People assume that
restarting the mon daemon should apply the change so my code
adds a function to check if the mon daemon public network
config has been changed and has not been applied yet. This
function is then called in _daemon_action to reroute a restart
request to the redeploy request process, essentially turning the
restart into a redeploy to update the public network.

Signed-off-by: Timothy Q Nguyen <timqn22@gmail.com>

adk3798 reviewed (commented):
Seems pretty close to being in a good spot, minor comments

adk3798 commented on src/pybind/mgr/cephadm/module.py:
```diff
@@ -2666,6 +2663,36 @@ def _rotate_daemon_key(self, daemon_spec: CephadmDaemonDeploySpec) -> str:
+                })
+
+        if rc_show:
+            self.log.error(f'cmd: config get failed with: {err_show}, (errno:{rc_show})')
```
some flake8 complaints as well
```
cephadm/module.py:2674:21: E126 continuation line over-indented for hanging indent
cephadm/module.py:2677:17: E126 continuation line over-indented for hanging indent
cephadm/module.py:2684:21: E126 continuation line over-indented for hanging indent
cephadm/module.py:2687:17: E126 continuation line over-indented for hanging indent
4     E126 continuation line over-indented for hanging indent
```

timqn22 replied:
I will fix.

timqn22 committed:
mgr/cephadm: hardcode daemon type to mon

Using the service_name variable is unnecessary due to
the output always being "mon", this is a safe change as
the function immediately checks that daemon type is
"mon". I included fixes for flake8, minor log
message error, and ensuring action is redeploy.

Signed-off-by: Timothy Q Nguyen <timqn22@gmail.com>

adk3798 reviewed (commented):
looks okay, just need to resolve our discussion on if we should alter the action

timqn22 commented:
jenkins test make check

rkachach commented:
Teuthology run was successful. Failing Test cases are not related to this PR: https://pulpito.ceph.com/rkachach-2026-05-11_13:27:21-orch:cephadm-wip-rkachach-testing-2026-05-11-1323-distro-default-trial/
